Clean up License, authors and contributing

jbreue16 opened the pull request:
This PR renames the third party license file for two reasons:
1) Github has trouble recognising our license when multiple file names start with LICENSE
2) Using the plural form since this file includes all third party licenses

I have tested this in a private repo and these changes should result in the following github toolbar
![image](https://github.com/user-attachments/assets/7a89746d-6d5c-467a-ab05-f162d788434d)

Fixes #322 in the above sense. The license is still not displayed as GPLv3.0 since this requires a completely unchanged license file.

hannahlanzrath commented:
We should still update the LICENSE.txt and create a NOTICE.txt according to #322 as well, so it doesn't just say "View License" but directly displays "GPL 3.0" in my opinion. Can be done in the same PR here.

jbreue16 commented:
We need to define the authors for the copyright and license.
Thats why I have added the authors file again, with a clear statement that defines the authors.
The contributing file is reduced to its designated purpose of providing information on contribution policy.

We should also add the years to the copyright statements in the source files again. We removed them to reduce maintenance, but i found that we can simply write 2004-present instead of specifying an end date that needs to be updated.

jbreue16 commented:
> We should still update the LICENSE.txt and create a NOTICE.txt according to #322 as well, so it doesn't just say "View License" but directly displays "GPL 3.0" in my opinion. Can be done in the same PR here.

I have no strong opinion on this. It would be neat to have the license displayed in the toolbar but its also nice to have copyright and license in one place, also when refering to it.

lieres commented:
I would prefer the license displayed in the sidebar.

jbreue16 commented on README.rst:
```diff
@@ -78,12 +78,26 @@ Donations
+----------------------------
 
-CADET-Core is licensed under the GPL v3.0 License (see `LICENSE.txt <https://github.com/cadet/cadet-core/blob/master/LICENSE.txt>`_).
+Copyright (C) 2008-present: The CADET-Core Authors (see `AUTHORS.md <https://github.com/cadet/cadet-core/blob/master/AUTHORS.md>`_).
```
wasnt it 2004?

jbreue16 commented:
@lieres does the copyright start at 2004 or 2008 ? Ive read both in different (and differently old) places

lieres commented:
Good question,. I think I coined the name CADET in 2008, while there was unnamed MATLAB software before.
